core/src/test/java/nl/nn/adapterframework/jdbc/JdbcTransactionalStorageTest.java

```diff
@@ -0,0 +1,216 @@
+package nl.nn.adapterframework.jdbc;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Date;
+import java.util.zip.DeflaterOutputStream;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import liquibase.Contexts;
+import liquibase.Liquibase;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.resource.FileSystemResourceAccessor;
+import nl.nn.adapterframework.stream.Message;
+import nl.nn.adapterframework.testutil.TestFileUtils;
+
+public class JdbcTransactionalStorageTest extends TransactionManagerTestBase {
+
+	private JdbcTransactionalStorage<Message> storage;
+	private Liquibase liquibase;
+	private final String tableName = "IBISSTORE_4_JdbcTransactionalStorageTest";
+	private final String messageField = "MESSAGE";
+	private final String keyField = "MESSAGEKEY";
+
+	@Before
+	public void setup() throws Exception {
+		super.setup();
+		storage = new JdbcTransactionalStorage<Message>();
+		storage.setTableName(tableName);
+		storage.setMessageField(messageField);
+		storage.setKeyField(keyField);
+		storage.setCheckTable(false);
+		storage.setDatasourceName(getDataSourceName());
+		storage.setDataSourceFactory(dataSourceFactory);
+		storage.setSequenceName("SEQ_IBISSTORE_4_JdbcTransactionalStorageTest");
+		createDbTable();
+	}
+
+	private void createDbTable() throws Exception {
+		FileSystemResourceAccessor resourceAccessor = new FileSystemResourceAccessor(TestFileUtils.getTestFileURL("/").getPath());
+		String changesetFilePath = TestFileUtils.getTestFileURL("/Migrator/Ibisstore_4_JdbcTransactionalStorageTest_changeset.xml").getPath();
```

Comment:
It might be a better idea to refer to the original IAF_DatabaseChangelog.xml, rather then making multiple copies that will get out of sync eventually.
It might also be a good idea to place this createDbTable method in JdbcTestBase.java